core/src/main/java/org/apache/calcite/adapter/jdbc/JdbcTable.java

```diff
@@ -100,6 +100,16 @@ public String toString() {
     return jdbcTableType;
   }
 
+  @Override public <C> C unwrap(Class<C> aClass) {
+    if (aClass.isInstance(jdbcSchema.getDataSource())) {
+      return aClass.cast(jdbcSchema.getDataSource());
+    } else if (aClass.isInstance(jdbcSchema.dialect)) {
+      return aClass.cast(jdbcSchema.dialect);
+    } else {
+      return super.unwrap(aClass);
+    }
+  }
+
   public RelDataType getRowType(RelDataTypeFactory typeFactory) {
     if (protoRowType == null) {
```

Comment:
Do you open a jira ticket for this? If you do, we can discuss it there.

Reply:
Here it is! https://issues.apache.org/jira/browse/CALCITE-3365